source/NPC.h

```diff
@@ -60,9 +60,13 @@ class NPC {
 	
 	// Handle the given ShipEvent.
 	void Do(const ShipEvent &event, PlayerInfo &player, UI *ui = nullptr, bool isVisible = true);
-	bool HasSucceeded(const System *playerSystem, bool checkDespawnConditions = true) const;
+	// Determine if the NPC is in a successful state, assuming the player is in the given system.
+	// (By default, a despawnable NPC has succeeded and is not actually checked.)
+	bool HasSucceeded(const System *playerSystem, bool ignoreIfDespawnable = true) const;
```

Comment:
Aren't we *not* ignoring if this is true? Variable name seems to imply the opposite of what the boolean does.

Reply:
I picked the name because of what it means for HasSucceeded:

- IF {{condition}} AND the NPC is despawnable, return TRUE, no matter what.
- IF {{condition}} but not despawnable, return based on full criteria evaluation
- IF NOT {{condition}}, return based on full criteria evaluation

I thought "we can ignore success criteria when despawnable" was a suitable substitution for {{condition}}, so I went with "ignoreIfDespawnable". "skipAllChecksIfDespawnable" would also work, but is rather lengthy.

Reply:
I don’t see why what I had didn’t work, as it describes exactly what is being done if the bool is true. (I.e. if checkDespawnConditions is true, we are checking passedDespawnConditions, but if it’s false then we’re not.) Could simply include a comment as to why we might want to not check if the despawn conditions have passed if the variable name wasn’t clear.

Reply:
The issue I had with `checkDespawnConditions` is that you can't infer from the name of the parameter what the semantics are. "Ok, if I send `checkDespawnConditions` as `true` to HasSucceeded, it will check the despawn conditions... but ... what does that mean about the result?"

Compare to "if I tell HasSucceeded to `ignoreIfDespawnable`, it's going to ignore {{things}} if the NPC is despawnable.

(An alternative name for this boolean parameter could have been `itsOkToLieToMeAboutTheConditions` -- if we really & truly evaluate the NPC's success & failure criteria, we might find that a despawnable NPC is in a failed state or hasn't otherwise succeeded. But, we're saying it's OK to lie about that for this call.)